assemble-and-restart

```diff
@@ -2,6 +2,34 @@
 set -x
 set -eo pipefail
 
+# If WorkingDir is injected as an env from odo and destination path is not equal to WorkingDir injected by odo
+if [ ! -z "${ODO_S2I_WORKING_DIR}" ] && [ "${ODO_S2I_SRC_BIN_PATH}" != "${ODO_S2I_WORKING_DIR}" ]; then
+    # If ODO_SRC_BACKUP_DIR is injected by odo
+    if [ -n  "$ODO_SRC_BACKUP_DIR" ]; then
+        # If it doesn't exit create it
+        if [ ! -d ${ODO_SRC_BACKUP_DIR} ]; then
+            mkdir -p ${ODO_SRC_BACKUP_DIR}/src
+        fi
+        # Backup the sources in DestinationDir to ODO_SRC_BACKUP_DIR bcoz the assemble script for some s2i images
+        # moves sources from DestinationDir to WorkingDir thereby deleting it there(DestinationDir). So, back it up
+        # for partial push use cases like watch
+        rsync -r ${ODO_S2I_SRC_BIN_PATH}/src/. ${ODO_SRC_BACKUP_DIR}/src/
+    fi
+    # Clear all those dirs in WorkingDir that exist in DestinationDir so that when assemble script moves sources from
+    # DestinationDir to WorkingDir, it sees the directory clean and doesn't complain:
+    # https://github.com/redhat-developer/odo/issues/1054
+    for file in `ls -a ${ODO_S2I_SRC_BIN_PATH}/src/`
```

Comment:
you can do `ls -A`  and then you don't need to check for `.` and `..` 

But problem is that this aproach with `for` and `ls` is not going to work properly when there is a filename that contains space!

Reply:
it would be good to have case with filename with space in our tests